### ISSUE TITLE
`manual_filter` suggestion broken on nested `if let`

### ISSUE THREAD

**Issue by niklasf** [rust-lang/rust-clippy#10088]
### Summary

When using nested `if let` in a way that triggers `clippy::manual_filter`, clippy makes an invalid suggestion.

### Reproducer

I tried this code:

```rust
fn maybe_some() -> Option<u32> {
    Some(0)
}

fn main() {
    let result = if let Some(a) = maybe_some() {
        if let Some(b) = maybe_some() {
            Some(a + b)
        } else {
            Some(a)
        }
    } else {
        None
    };
    
    println!("{result:?}");
}
```

I expected to see this happen: No lint error, or suggested fix compiles.

Instead, this happened:

```
warning: manual implementation of `Option::filter`
  --> src/main.rs:6:17
   |
6  |       let outer = if let Some(a) = maybe_some() {
   |  _________________^
7  | |         if let Some(b) = maybe_some() {
8  | |             Some(a + b)
9  | |         } else {
...  |
13 | |         None
14 | |     };
   | |_____^ help: try this: `maybe_some().filter(|&a| !(let Some(b) = maybe_some()))`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_filter
   = note: `#[warn(clippy::manual_filter)]` on by default
```

### Version

```text
0.1.67 (2022-12-14 b70baa4)
```

**Comment by EricWu2003** [comment]
I'm currently trying to tackle this issue and I realized that the code below also causes clippy to emit a code suggestion that is not correct:
```rust
fn main() {
    match Some(0) {
        None => None,
        Some(x) => {
            if x > 0 {
                Some(3*x)
            } else {
                Some(x)
            }
        },
    };
}
```
Here clippy gives the suggestion:
```text
warning: manual implementation of `Option::filter`
  --> src/main.rs:6:5
   |
6  | /     match Some(0) {
7  | |         None => None,
8  | |         Some(x) => {
9  | |             if x > 0 {
...  |
14 | |         },
15 | |     };
   | |_____^ help: try this: `Some(0).filter(|&x| x <= 0)`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_filter
```
